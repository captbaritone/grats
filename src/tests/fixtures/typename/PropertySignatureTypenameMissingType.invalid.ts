/** @gqlType */
export class User implements IPerson {
  __typename;
  /** @gqlField */
  name: string;
}

/** @gqlInterface */
export interface IPerson {
  /** @gqlField */
  name: string;
}
