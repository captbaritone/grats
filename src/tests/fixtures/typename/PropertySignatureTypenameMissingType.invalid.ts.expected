-----------------
INPUT
----------------- 
/** @gqlType */
export class User implements IPerson {
  __typename;
  /** @gqlField */
  name: string;
}

/** @gqlInterface */
export interface IPerson {
  /** @gqlField */
  name: string;
}

-----------------
OUTPUT
-----------------
-- Error Report --
src/tests/fixtures/typename/PropertySignatureTypenameMissingType.invalid.ts:3:3 - error: Expected `__typename` property to have an initializer or a string literal type.  This is needed to ensure Grats can determine the type of this object during GraphQL execution.

3   __typename;
    ~~~~~~~~~~


-- Code Action: "Create Grats-compatible `__typename` property" (fix-typename-property) --
- Original
+ Fixed

@@ -2,3 +2,3 @@
  export class User implements IPerson {
-   __typename;
+   __typename = "User" as const;
    /** @gqlField */